types/src/crypto.rs

```diff
@@ -33,3 +40,21 @@ pub fn blake2b<T: AsRef<[u8]>>(data: T) -> [u8; BLAKE2B_DIGEST_LENGTH] {
     });
     result
 }
+
+/// A type of hashing algorithm.
+#[repr(u8)]
+#[derive(Debug, Copy, Clone, PartialEq, Eq, FromPrimitive)]
+pub enum HashAlgoType {
+    /// Blake2b
+    Blake2b = 0,
+    /// Blake3
+    Blake3 = 1,
+}
+
+impl TryFrom<u8> for HashAlgoType {
+    type Error = bytesrepr::Error;
+
+    fn try_from(value: u8) -> Result<Self, bytesrepr::Error> {
+        FromPrimitive::from_u8(value).ok_or(bytesrepr::Error::Formatting)
```

Comment:
I'd probably use `ApiError::InvalidArgument` here to make it more obvious, since `Formatting` error may be confusing as there's no serialization involved

Reply:
1. I agree `Formatting` error might be confusing. Why I did that? I based on [`ExecutionResultTag`](https://github.com/casper-network/casper-node/blob/v1.5.3/types/src/execution_result.rs#L44-L57) and few other enums in `execution_result.rs`, but now I see those `TryFrom`'s are always used internally inside `ExecutionResult::from_bytes()`.

2. I think `ApiError` should not be present in base Casper types, as no API is involved here. Therefore I think we can just **keep it simple** and make `()` the error:

![image](https://github.com/casper-network/casper-node/assets/6506780/6d9cdd60-3026-46af-9005-5c3966bd28a4)

3. We can return `ApiError::InvalidArgument` inside function handler - it makes sense - but it complicates the code a little bit. Nevertheless, done in https://github.com/casper-network/casper-node/pull/4412/commits/d064e2c90148a5e13b3ecaf1f98a1f3646e04ee0.

Reply:
Additional, please take in mind that user will not able to face this error, because using `HashAlgoType` enum as an argument forces the correct value.